Content.Server/Administration/QuickDialogSystem.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System.ComponentModel;
+using Content.Shared.Administration;
+using Robust.Server.Player;
+using Robust.Shared.Enums;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+
+namespace Content.Server.Administration;
+
+/// <summary>
+/// This handles the server portion of quick dialogs, including opening them.
+/// </summary>
+public sealed partial class QuickDialogSystem : EntitySystem
+{
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+
+    /// <summary>
+    /// Contains the success/cancel actions for a dialog.
+    /// </summary>
+    private readonly Dictionary<int, (Action<QuickDialogResponseEvent> okAction, Action cancelAction)> _openDialogs = new();
+    private readonly Dictionary<NetUserId, List<int>> _openDialogsByUser = new();
+
+    private int _nextDialogId = 1;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        _playerManager.PlayerStatusChanged += PlayerManagerOnPlayerStatusChanged;
```

Comment:
unsubscribe on shutdown.

Technically this is only required on client but I'd rather just do it now as we see it and not need to do it later if we need to tbh.